src/saveload/station_sl.cpp

```diff
@@ -224,6 +231,36 @@ class SlStationSpecList : public DefaultSaveLoadHandler<SlStationSpecList, BaseS
 
 uint8 SlStationSpecList::last_num_specs;
 
+class SlRoadStopSpecList : public DefaultSaveLoadHandler<SlRoadStopSpecList, BaseStation> {
+public:
+	inline static const SaveLoad description[] = {
+		SLE_CONDVAR(RoadStopSpecList, grfid,    SLE_UINT32, SLV_NEWGRF_ROAD_STOPS, SL_MAX_VERSION),
+		SLE_CONDVAR(RoadStopSpecList, localidx, SLE_UINT8,  SLV_NEWGRF_ROAD_STOPS, SL_MAX_VERSION),
+	};
+	inline const static SaveLoadCompatTable compat_description = _station_road_stop_spec_list_sl_compat;
+
+	void Save(BaseStation *bst) const override
+	{
+		SlSetStructListLength(bst->roadstop_speclist.size());
+		for (uint i = 0; i < bst->roadstop_speclist.size(); i++) {
+			SlObject(&bst->roadstop_speclist[i], this->GetDescription());
+		}
+	}
+
+	void Load(BaseStation *bst) const override
+	{
+		uint8 num_specs = 0;
+		if (!IsSavegameVersionBefore(SLV_NEWGRF_ROAD_STOPS)) {
+			num_specs = (uint8)SlGetStructListLength(UINT8_MAX);
+		}
```

Comment:
This doesn't need to be conditional, this chunk will always have the struct list length here. It's conditional for rail station speclist only because that used to be stored separately.